src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoaderExtensions.cs

```diff
@@ -16,13 +16,45 @@
 
 using System;
 using System.IO;
+using System.Linq;
+using System.Reflection;
 using System.Text;
+using Nethermind.Config;
 using Nethermind.Logging;
 
 namespace Nethermind.Specs.ChainSpecStyle
 {
     public static class ChainSpecLoaderExtensions
     {
+
+        public static ChainSpec LoadFromEmbeddedOrFile(this IChainSpecLoader chainSpecLoader, string fileName, ILogger logger)
+        {
+            string resourceName = fileName;
+            if (!resourceName.Contains('/'))
+            {
+                resourceName = "chainspec/" + resourceName;
+            }
+            if (!resourceName.EndsWith(".json", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Should be `StringComparison.OrdinalIgnoreCase` instead of `StringComparison.InvariantCultureIgnoreCase`. It's faster. We have this in other places as well but that's not a part of this PR.